### ISSUE TITLE
Display on pone weapons or armor

### ISSUE THREAD

**Issue by r0m1ntik** [azerothcore/azerothcore-wotlk#4878]
##### CURRENT BEHAVIOUR:

When you have a pair of identical weapons or equipments, apply different enchantments to them, add them to your panel.
Ano will count as if it is 1 and the same item, when you click on them, it will equip 1 and the same item, even if you put 3 with different charms.

[https://www.youtube.com/watch?v=2tZ6Fdio0lg](https://www.youtube.com/watch?v=2tZ6Fdio0lg)

##### EXPECTED BLIZZLIKE BEHAVIOUR:

As for me, it shouldn't be so.

##### STEPS TO REPRODUCE THE PROBLEM:

1. give yourself 2 or more identical items
2. enchant them differently or put a socket
3. insert them into the panel and see what they will show

##### AC HASH/COMMIT:

[d20a15355c1aca4942a726d2092c66918b94e9ee](https://github.com/azerothcore/azerothcore-wotlk/commit/d20a15355c1aca4942a726d2092c66918b94e9ee)

##### OPERATING SYSTEM:

Ubuntu 18

**Comment by Eureka26** [comment]
i can confirm this, i also tried on Trinity-core and they have same issue.

**Comment by esanhueza** [comment]
I ran into some problems when I tried to fix this issue. I noticed that the server only receives the Item ID and not the ID of the instance of that item. That means the server doesn't really know which copy is the one being attached to the action bar.

**Comment by r0m1ntik** [comment]
@esanhueza please tell me where did you find this

**Comment by esanhueza** [comment]
@r0m1ntik 
https://github.com/azerothcore/azerothcore-wotlk/blob/60f02f86d6c9d2eba0a0e6be05ecf264a6cc860c/src/server/game/Handlers/MiscHandler.cpp#L907
That's the function that handles the event of adding an action to the action bar.

**Comment by r0m1ntik** [comment]
@esanhueza 
In character_action, the ID of the subject is written, I think it is necessary to write down the GUID there and rewrite the receipt of information, but I may be wrong, while there are no other ideas.

**Comment by esanhueza** [comment]
@r0m1ntik even if the server sends the GUID, the client needs the EntryID. You can check it here [https://github.com/azerothcore/azerothcore-wotlk/blob/60f02f86d6c9d2eba0a0e6be05ecf264a6cc860c/src/server/game/Entities/Player/Player.cpp#L6827](url). Actions are loaded and sent directly to the client.
Could it be that there is another ActionButtonType to differentiate between the elements the player has and the others? I don't know how to check it.

**Comment by Shard-MW** [comment]
I don't think that this issue comes from server, but mostly from 3.3.5a client.
Is there any proof (video, screen...) anywhere that this worked on retail ? 
Because if client sends ItemID to the server and not the GUID, it means that the client was coded like this

**Comment by r0m1ntik** [comment]
@Shard-MW is not a client problem, on warmane it's work.

**Comment by Tolky** [comment]
@r0m1ntik I just tested it on Warmane (Blackrock server) and it doesn't work. Still get the same weapon in action bars.
